diff -ur Python-2.1a1.old/Lib/cgi.py Python-2.1a1/Lib/cgi.py
--- Python-2.1a1.old/Lib/cgi.py	Sat Jan 20 12:54:20 2001
+++ Python-2.1a1/Lib/cgi.py	Fri Jan 26 01:07:57 2001
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#!/usr/bin/env python
 
 """Support module for CGI (Common Gateway Interface) scripts.
 
diff -ur Python-2.1a1.old/Tools/faqwiz/faqw.py Python-2.1a1/Tools/faqwiz/faqw.py
--- Python-2.1a1.old/Tools/faqwiz/faqw.py	Fri Apr  3 15:27:04 1998
+++ Python-2.1a1/Tools/faqwiz/faqw.py	Fri Jan 26 01:08:07 2001
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#!/usr/bin/env python
 
 """FAQ wizard bootstrap."""
 
diff -ur Python-2.1a1.old/Tools/scripts/parseentities.py Python-2.1a1/Tools/scripts/parseentities.py
--- Python-2.1a1.old/Tools/scripts/parseentities.py	Wed Jan 17 01:48:39 2001
+++ Python-2.1a1/Tools/scripts/parseentities.py	Fri Jan 26 01:08:16 2001
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/env python
 """ Utility for parsing HTML entity definitions available from:
 
       http://www.w3.org/ as e.g.
