diff -ur tix-8.1.4/win/common.mak tix-8.1.4.new/win/common.mak
--- tix-8.1.4/win/common.mak	2002-12-11 07:19:42.000000000 +0100
+++ tix-8.1.4.new/win/common.mak	2004-08-03 21:45:09.859375000 +0200
@@ -18,10 +18,10 @@
 #	    support files
 #
 #----------------------------------------------------------------------
-TCL_VER          = 8.3
+TCL_VER          = 8.4
 ITCL_VER          = 
 
-INSTALLDIR      = C:\progra~1\tcl
+INSTALLDIR      = ..\..\tcltk
 
 !IFNDEF TIX_DEBUG
 NODEBUG = 1
@@ -61,7 +61,7 @@
 !IF "$(TCL_VER)" == "8.4"
 TCLMAJOR=8
 TCLMINOR=4
-TCLPATCH=1
+TCLPATCH=7
 TMPDIR          = tk$(TCL_VER)
 !ENDIF
 
@@ -176,14 +176,14 @@
 	$(TMPDIR)\tixWinWm.obj
 
 RMDIR		= $(TCLDIR)\win\rmd.bat
-MKDIR		= $(TCLDIR)\win\mkd.bat
+MKDIR		= mkdir
 RM		= del
 
 install:    install-binaries install-libraries
 
 install-binaries: $(TCLSH)
-	$(MKDIR) "$(BIN_INSTALL_DIR)"
-	$(MKDIR) "$(LIB_INSTALL_DIR)"
+	-$(MKDIR) "$(BIN_INSTALL_DIR)"
+	-$(MKDIR) "$(LIB_INSTALL_DIR)"
 	@echo installing $(TIXDLL)
 	@copy "$(TIXDLL)" "$(BIN_INSTALL_DIR)"
 	@copy "$(TIXLIB)" "$(LIB_INSTALL_DIR)"
diff -ur tix-8.1.4/win/makefile.vc tix-8.1.4.new/win/makefile.vc
--- tix-8.1.4/win/makefile.vc	2002-12-02 04:02:54.000000000 +0100
+++ tix-8.1.4.new/win/makefile.vc	2004-08-03 21:42:07.953125000 +0200
@@ -54,12 +54,11 @@
 DBGX = d
 !ENDIF
 
-cc32   = "$(TOOLS32)\bin\cl.exe"
-rc32   = "$(TOOLS32_rc)\bin\rc.exe"
-link32 = "$(TOOLS32)\bin\link.exe"
-include32 = -I"$(TOOLS32)\include"
+cc32   = "cl.exe"
+rc32   = "rc.exe"
+link32 = "link.exe"
 
-TIX_INCLUDES = $(include32) \
+TIX_INCLUDES = \
 	-I$(ROOT)\win -I$(ROOT)\generic \
 	-I$(TKDIR)\generic -I$(TKDIR)\win -I$(TKDIR)\xlib \
 	-I$(TCLDIR)\generic $(ITCL_CFLAGS)
@@ -171,7 +170,7 @@
 #
 cvarsdll  = -D_X86_=1 -DWIN32 -D_WIN32 -D_MT -D_DLL
 cflagsdll = $(cvarsdll) -c -W3 -nologo -Fp$(TMPDIR)\ -YX -MD \
-	    -Oti -Gs -GD
+	    -Oti -Gs -Gd
 
 ######################################################################
 # Project specific targets
@@ -181,7 +180,6 @@
 
 $(DUMPEXTS): $(WINDIR)\winDumpExts.c
 	$(cc32) $(CON_CFLAGS) -Fo$(TMPDIR)\ /c $?
-	set LIB="$(TOOLS32)\lib"
 	$(link32) $(ldebug) $(conlflags) $(guilibs) -out:$@ \
 		$(TMPDIR)\winDumpExts.obj 
 
@@ -193,7 +191,6 @@
 # (ToDo) $(TIXDLL) doesn't have resources to define its icon, etc.
 #
 $(TIXDLL): $(TIXOBJS) $(TMPDIR)\tixvc.def
-	set LIB="$(TOOLS32)\lib"
 	$(link32) $(ldebug) $(dlllflags) -def:$(TMPDIR)\tixvc.def \
 		$(TKLIBDIR)\$(TKLIB) $(TCLLIBDIR)\$(TCLLIB) $(guilibsdll) \
 		$(ITCL_LIBS) -out:$@ @<<
@@ -202,7 +199,6 @@
 
 
 $(TIXWISH): $(WISHOBJS) $(TIXOBJS) $(TIXLIB) $(TMPDIR)\tixwish.res
-	set LIB="$(TOOLS32)\lib"
 	$(link32) $(ldebug) $(guilflags) \
 		$(WISHOBJS) $(TMPDIR)\tixwish.res $(TIXLIB) \
 		$(TKLIBDIR)\$(TKLIB) $(TCLLIBDIR)\$(TCLLIB) $(guilibsdll) \
diff -ur tix-8.1.4/win/tk8.4/pkgIndex.tcl tix-8.1.4.new/win/tk8.4/pkgIndex.tcl
--- tix-8.1.4/win/tk8.4/pkgIndex.tcl	2002-12-15 04:21:54.000000000 +0100
+++ tix-8.1.4.new/win/tk8.4/pkgIndex.tcl	2004-08-31 08:38:43.921875000 +0200
@@ -15,7 +15,7 @@
 # We look in the ../../bin directory (an installed Tcl)
 lappend dirs ../../bin
 # We look in the ../../DLLs directory (an installed Python)
-lappend dirs ../../Dlls
+lappend dirs [file join [file dirname [info nameofexe]] DLLs]
 # If not, this pkgIndex.tcl will probably fail.
 
 
